tests/functional/aws-node-sdk/test/object/deleteObject.js

```diff
@@ -13,70 +22,218 @@ describe('DELETE object', () => {
+                return s3.createBucketPromise({
+                    Bucket: bucketName,
+                    ObjectLockEnabledForBucket: true,
+                })
+                .catch(err => {
+                    process.stdout.write(`Error creating bucket ${err}\n`);
+                    throw err;
+                })
+                .then(() => {
+                    process.stdout.write('putting object\n');
+                    return s3.putObjectPromise({
+                        Bucket: bucketName,
+                        Key: objectName,
+                    });
+                })
+                .catch(err => {
+                    process.stdout.write('Error putting object');
+                    throw err;
+                })
+                .then(res => {
+                    versionIdOne = res.VersionId;
+                    process.stdout.write('putting object retention\n');
+                    return s3.putObjectRetentionPromise({
+                        Bucket: bucketName,
+                        Key: objectName,
+                        Retention: {
+                            Mode: 'GOVERNANCE',
+                            RetainUntilDate: retainDate,
+                        },
+                    });
+                })
+                .catch(err => {
+                    process.stdout.write('Err putting object retention\n');
+                    throw err;
+                });
+                // see comment on line 222 for why this setup is commented out
+                // .then(() => {
+                //     process.stdout.write('putting object\n');
+                //     return s3.putObjectPromise({
+                //         Bucket: bucketName,
+                //         Key: objectNameTwo,
+                //     });
+                // })
+                // .catch(err => {
+                //     process.stdout.write(('Err putting second object\n'));
+                //     throw err;
+                // })
+                // .then(res => {
+                //     versionIdTwo = res.VersionId;
+                //     process.stdout.write('putting object legal hold\n');
+                //     return s3.putObjectLegalHoldPromise({
+                //         Bucket: bucketName,
+                //         Key: objectNameTwo,
+                //         LegalHold: {
+                //             Status: 'ON',
+                //         },
+                //     });
+                // })
+                // .catch(err => {
+                //     process.stdout.write('Err putting object legal hold\n');
+                //     throw err;
+                // });
+            });
+
+            after(() => {
+                // process.stdout.write('removing object legal hold\n');
+                // return metadataGetObjectPromise(
+                //     bucketName, objectNameTwo, versionIdTwo, log)
+                // .then(objMDTwo => {
+                //     objMDTwo.setLegalHold(false);
```

Comment:
I think removing these commented out blocks (maybe moving them into a ticket for now) could make the file look cleaner.